dataclients/kubernetes/kube.go

```diff
@@ -349,14 +349,25 @@ func (c *Client) convertDefaultBackend(i *ingressItem) (*eskip.Route, bool, erro
 	return r, true, nil
 }
 
-func (c *Client) convertPathRule(ns, name, host string, prule *pathRule) (*eskip.Route, error) {
+func (c *Client) convertPathRule(ns, name, host string, prule *pathRule, servicesURLs map[string]string) (*eskip.Route, error) {
 	if prule.Backend == nil {
 		return nil, fmt.Errorf("invalid path rule, missing backend in: %s/%s/%s", ns, name, host)
 	}
-
-	address, err := c.getServiceURL(ns, prule.Backend.ServiceName, prule.Backend.ServicePort)
-	if err != nil {
-		return nil, err
+	serviceKey := ns + prule.Backend.ServiceName + prule.Backend.ServicePort.String()
```

Comment:
i was thinking about cases where we have a) namespace=foo,service=barbaz and b) namespace=foobar,service=baz . I know it doesn't eliminate the conflicts completely, only reduces the chance. For that we'd need to escape somehow each occurence of __ in the components.

I just checked what we do with the routeIDs, and see now that those are also not completely right, because there the '_' occurences are simply deleted. This has the potential of some rare but annoying bugs.

We can fix this for both the service url keys and route IDs as a separate issue, so I am not insisting on doing it here.